### ISSUE TITLE
Issues/PRs not reloaded before data gets fetched when changing repository from header

### ISSUE THREAD

**Issue by Echomo-Xinyu** [CATcher-org/WATcher#100]
**Describe the bug**

When changing repository from the header, the original repository is as below:

<img width="1311" alt="Screenshot 2023-03-20 at 12 45 04 PM" src="https://user-images.githubusercontent.com/25452497/226249290-77f22e25-1b43-4c05-9b1f-1e759b9c93c3.png">

But after changing to another repository from the header, 

<img width="430" alt="image" src="https://user-images.githubusercontent.com/25452497/226251652-d6c61f7d-5959-4185-8928-4c70871cd7c4.png">

<img width="801" alt="image" src="https://user-images.githubusercontent.com/25452497/226251702-d0a38b6c-efa3-497f-8389-a51f964ad433.png">

the issues and PRs do not reload immediately

<img width="1391" alt="image" src="https://user-images.githubusercontent.com/25452497/226251737-bf2e1115-6bd8-4e82-8226-7317c2e06658.png">

the issues and PRs displayed will only get updated when the data with the new repository gets fetched from GitHub

<img width="1451" alt="image" src="https://user-images.githubusercontent.com/25452497/226251813-472f2f9f-dfaf-4753-ada4-295b4801ba18.png">

**To Reproduce**

1. Go to 'AY2223S1-CS2103T-T09-4/tp' on WATcher
2. Click on 'Change Repository' on header
3. Input  'CATcher-org/WATcher'

**Expected behavior**

The issues and PRs should get reloaded automatically instead of being displayed while fetching data. A loading screen should be displayed during the fetching.

**Desktop (please complete the following information):**

- OS: MacOS Monterey (12.5)
- Browser Chrome 110
- Version 110.0.5481.177

**Comment by vigneshsankariyer1234567890** [comment]
I think there's actually a couple of issues here:
1. There isn't actually an indication to see if any requests are loading or not (like a spinner when we first login into WATcher)
2. Items are not being cleared from memory 

For the second issue, I've found the reason for the issue. Here is the flow of events that occur in WATcher when the repo is changed after opening the Change Repo Dialog:
1. `PhaseService::changeCurrentRepository` is called on a new `Repo` instance consisting of the new repository address provided by the user
2. This results in `PhaseService` calling `next()` on its `repoChanged$` Subject that stores information on the last repository which was changed. The new repo instance is passed to `next()`
3. Since `IssuesViewerComponent` has a subscription on `PhaseService::repoChanged$`, when the new repo is passed to `next` as in 2, the callback to `IssuesViewerComponent::initialize` is subsequently called.
4. `IssuesViewerComponent::initialize` then resets its assignees and reloads all Issues using `IssueService::reloadAllIssues`.
5. The call to `IssueService::reloadAllIssues` results in a call to `IssueService::initializeData`, which always results in this call: `this.githubService.fetchIssuesGraphql(new RestGithubIssueFilter({}))`. This is because the filter would always point to 'NO_FILTER' since WATcher is not dependent on phases and does not change the filter. However, WATcher does not remove all issues from its local store of issues before fetching issues from GitHub, but rather fetches the issues and then deletes the issues which are not supposed to be in the new set of fetched issues. This results in the persistence of issues, especially when the 2 repositories or teams have common members. 

I think we can implement an easy fix by simply flushing the store first and then fetching and setting issues. There is also a method in `issueService` that resets the issue in the service to an empty array and does the action above. This could kill 2 birds with one stone as it also sets the loading variable to true and ultimately displays the spinners in their appropriate location. 

Seems like a pretty complicated problem given the current flow. WDYT @CATcher-org/2223s2
